udfs/migration/teradata/left.sql

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+CREATE OR REPLACE FUNCTION td.left(string_expr STRING, length_expr INT64) AS (
+  SUBSTR(string_expr, 1, length_expr)
```

Comment:
I would handle the negative-case because `SUBSTR` will throw an error with a negative `length_expr` and it may be hard for the user to debug.